src/OSPSuite.Core/Domain/Services/ConcurrencyManager.cs

```diff
@@ -52,5 +88,37 @@ public class ConcurrencyManager : IConcurrencyManager
          var tt = results.Values;
          return results;
       }
+
+      private async Task<ConcurrencyManagerResult<TResult>> returnWithExceptionHandling<TData, TResult>
+      (
+         int coreId, 
+         CancellationToken cancellationToken, 
+         Func<int, CancellationToken, TData, Task<TResult>> task, 
+         TData args, Func<TData, string> id
+      ) where TResult : class
+      {
+         TResult simulationResult;
+         try
+         {
+            simulationResult = await task.Invoke(coreId, cancellationToken, args);
+         }
+         catch (Exception e)
+         {
+            return new ConcurrencyManagerResult<TResult>
+            (
+               id(args),
+               false,
+               null,
+               e.Message
+            );
+         }
+         return new ConcurrencyManagerResult<TResult>
```

Comment:
why don't you return this from the try? 
Also can you not name the parameter `args`
Last, don't hesitate to name parameters instead of passing false, or true => success: true